assert.go

```diff
@@ -240,6 +257,26 @@ func assertEqual(t testingT, expr func() string, got, want interface{}, opts []c
 	return true
 }
 
+func assertNotEqual(t testingT, expr func() string, got, notWant interface{}, opts []cmp.Option) bool {
+	defer func() {
+		if err := recover(); err != nil {
+			t.Error("diff error:", err)
+		}
+	}()
+	t.Helper()
+	opts = append(opts, defaultOpts...)
+	if diff := cmp.Diff(got, notWant, opts...); diff == "" {
+		expr := expr()
+		msg := fmt.Sprintf("got %#v", got)
```

Comment:
Hmm. Maybe: 
```suggestion
		msg := fmt.Sprintf("should not equal %#v", notWant)
```
so that an error might be like
> subject: should not equal "subject"`